### DISCUSSION TITLE
ipython integration

### DISCUSSION THREAD

**Discussion by bannsec** [community/community#6922] [Copilot Conversations]
Understanding that it's not on the cards rn, but it would be pretty nice if copilot could be integrated directly into the ipython repl. it could take the current state of the interpreter into account and provide suggestions as you're coding in it.

**Comment by Krzysztof-Cieslak** [comment]
Hey,

While this is not 100% what you are asking about, I'd like to point out we do support Jupyter Notebooks opened in VSCode (with Jupyter extension).

**Reply by Anselmoo** [reply]
@Krzysztof-Cieslak in addition to @bannsec, is there any plans to integrate copilot into the terminal ipython?

<img width="651" alt="image" src="https://user-images.githubusercontent.com/13209783/165251741-0c9c32d8-25ed-4cd9-9966-deed04891991.png">

As I can see, you are already providing an amazing `vim-plugin` for __Copilot__ [https://github.com/github/copilot.vim](https://github.com/github/copilot.vim)
